client/src/screens/group.screen.js

```diff
@@ -94,41 +178,131 @@ class Group extends Component {
     ),
   };
 
-  keyExtractor = item => item.id;
+  onRefresh() {
+    this.props.refetch();
+  }
 
-  renderItem = ({ item }) => <Member member={item} />
+  joinOrLeaveGroup(group, memberAlready) {
+    const { joinGroup, leaveGroup, refetch } = this.props;
+    if (!memberAlready) {
+      joinGroup({ groupId: group.id }).then(() => {
+      // dont need to do anything, but we might in the future
+      }).catch((error) => {
+        Alert.alert(
+          'Error Joining Group',
+          error.message,
+          [
+            { text: 'OK', onPress: () => {} },
+          ],
+        );
+      });
+    } else {
+      leaveGroup({ groupId: group.id }).then(() => {
+        // pull group details down again to populate the group details
+        refetch();
```

Comment:
the fact you have to do this probably indicates something isn't set up quite right but it's probably not worth the time investment to investigate right now.

Reply:
I can either return the whole group when you leave and pass that back to the page (which would need a new query in the remove function because it returns a int of the number of rows removed, not an object), or trigger the page to refresh which will fetch the group again and update the cache by ID's for all linked items. Both ways need a query, this way is just easier on the cache